### DISCUSSION TITLE
cy.intercept() with query parameters

### DISCUSSION THREAD

**Discussion by bartbriek** [cypress-io/cypress#9318] [Questions and Help]
Hi,

I had the following code in one of my tests.

```
cy.server();
cy.route({
      url:`/organizations/${userId}?query=Te`, 
      response: 'fx:/organizations/organizationsearchform.json'
    }).as('results');
```
What would be the equivalent with the new cy.intercept() function?

```
cy.intercept({
      method: 'GET',
      url: `/organizations/${userId}`,
      query: { query: 'Te' },
      response: { fixture: '/organizations/organizationsearchform.json' }
    }).as('results');
```

I tried this, but cypress is not picking up the query parameters and is giving me a response for ```cy.intercept(`/organizations/${userId}`, { body: [] }).as('organizations');```

Thanks in advance.

**Comment by bahmutov** [comment]
Can you try `pathname` instead of the `url`? See the examples I have added in https://github.com/cypress-io/cypress-example-recipes/pull/592

**Reply by bartbriek** [reply]
Thanks for the examples. I tried the option with pathname as well, but it had the same effect. I narrowed it down to the following:

when 
```
cy.intercept(`/organizations/${userId}`, { body: [] }).as('organizations');
``` 
is put before 

``` 
cy.intercept({
      method: 'GET',
      url: `/organizations/${userId}`,
      query: { query: 'Te' },
      response: { fixture: '/organizations/organizationsearchform.json' }
    }).as('results');
```
it doesn't find the results alias.

when organizations (alias route) is put after results, cypress recognizes the results alias. But also gives organizations back in the req. As organizations is put last that is the response i get on the query route. 

![image](https://user-images.githubusercontent.com/39011473/100138100-cbd60780-2e8d-11eb-9951-50582c5458a6.png)

![image](https://user-images.githubusercontent.com/39011473/100140109-caf2a500-2e90-11eb-96a7-556e34b223ee.png)

![image](https://user-images.githubusercontent.com/39011473/100138537-79491b00-2e8e-11eb-8926-aa683415b8b9.png)

Hope this makes sense.
